### ISSUE TITLE
Why must strings be quoted?

### ISSUE THREAD

**Issue by KallDrexx** [kdl-org/kdl#246]
Is there a reason that single word strings must be quoted?

I was excited about KDL as a configuration format because it seemed like something that will make configurations easier for non-engineers than YAML or JSON.  However, I am concerned about the requirement for strings to be quoted. 

For example, in my application I had the following config I was trying to test with:

```
settings {
    feature_a on
    feature_b restricted_mode
}
```

This fails to parse in spec compliant parsers which made me really confused.  It seems to have failed because `on` and `restricted_mode` both need to be quoted for it to work.  However, how am I supposed to effectively communicate that to users?

Parsers still must do multiple passes over a single argument, so I am having a tough time seeing the benefit of this requirement.  For example, each argument I would need to see if it starts with a quote, is it `null`, is it a number, if it's a number does it have a decimal place, etc...  It seems to me that if it's an unquoted value that's not `null`, and not a number it should fall back to a string.

**Comment by eugenesvk** [comment]
> KDL has less keywords (only true/false/null), but people run into issues with YAML parsing even for those shared keywords, where they unexpectedly get a keyword rather than the string they intended. See also things like 1.1 parsing as a number but 1.1.1 parsing as an unquoted string; type confusion, again, but also guess what happens when you write version 1.10!

Isn't this neatly solved via using syntax highlighting? Then `1.1` will have different color vs `1.1.1`, so the feedback is immediate, and then the color change again when you do `1.10`?

**Comment by tabatkins** [comment]
The fact that people keep committing the error in YAML suggests that no, syntax highlighting is not a sufficient guard against misinterpretation. (The obvious reason why not is that not everyone uses syntax highlighting on every file, in every program.)
